microsoft/testsuites/storage/performance.py

```diff
@@ -0,0 +1,189 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+from enum import Enum
+from typing import Any, Dict, List
+
+from assertpy import assert_that
+
+from lisa import (
+    Logger,
+    Node,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    notifier,
+    schema,
+    search_space,
+    simple_requirement,
+)
+from lisa.environment import Environment
+from lisa.features import Disk
+from lisa.notifier import FIOResult, FIOResultsMessage
+from lisa.tools import Fdisk, Fio, Lscpu, Mdadm
+
+FIOMODES = Enum(
+    "FIOMODES",
+    [
+        "randread",
+        "randwrite",
+        "read",
+        "write",
+    ],
+)
+
+
+def _format_disk(
+    node: Node,
+    disk_list: List[str],
+) -> List[str]:
+    fdisk = node.tools[Fdisk]
+    partition_disks: List[str] = []
+    for data_disk in disk_list:
+        fdisk.delete_partitions(data_disk)
+        partition_disks.append(fdisk.make_partition(data_disk, format=False))
+    return partition_disks
+
+
+def _stop_raid(node: Node) -> None:
+    mdadm = node.tools[Mdadm]
+    mdadm.stop_raid()
+
+
+def _make_raid(node: Node, disk_list: List[str]) -> None:
+    mdadm = node.tools[Mdadm]
+    mdadm.create_raid(disk_list)
+
+
+@TestSuiteMetadata(
+    area="storage",
+    category="performance",
+    description="""
+    This test suite is to validate premium SSD data disks performance of Linux VM using
+     fio tool.
+    """,
+)
+class storageperf(TestSuite):  # noqa
+    TIME_OUT = 3000
+
+    @TestCaseMetadata(
+        description="""
+        This test case uses fio to test data disk performance with 4K block size.
+        """,
+        priority=2,
+        timeout=TIME_OUT,
+        requirement=simple_requirement(
+            min_core_count=72,
+            disk=schema.DiskOptionSettings(
+                disk_type=schema.DiskType.PremiumSSDLRS,
+                data_disk_iops=5000,
+                data_disk_count=search_space.IntRange(min=16),
+            ),
+        ),
+    )
+    def perf_premium_datadisks_4k(self, node: Node, environment: Environment) -> None:
+        self._perf_premium_datadisks(node, environment)
+
+    @TestCaseMetadata(
+        description="""
+        This test case uses fio to test data disk performance using 1024K block size.
+        """,
+        priority=2,
```

Comment:
Does it need to be p2? It's a perf test case. what's the original priority in v2?

Reply:
original priority in v2 is 1

Reply:
Let's move all perf test cases to p3, unless it can be done in 5 minutes.

Reply:
ok

Reply:
updated